lib/_http_client.js

```diff
@@ -197,7 +197,9 @@ function ClientRequest(options, cb) {
     self = null;
   });
 
-  this._ended = false;
+  self._ended = false;
```

Comment:
This change doesn't affect anything, but I found two different ways across the single constructor: `self` and `this`. Seems it should be consistent but I don't know the agreement about it.